src/stores/database.js

```diff
@@ -115,6 +116,14 @@ class Database {
     await umzug.up();
   }
 
+  async generateSchemaHash() {
+    const migrations = await this._db.collection('migrations').find({}).toArray();
+    const migrationNames = migrations.map(migration => migration.name).sort().join();
+    this.schemaHash = md5(migrationNames);
```

Comment:
I think we should create the schema hash from the migration names in the app, not query it from the DB. What do you think?

Reply:
They should be the same... it seems simpler to me to take them form the db.

Reply:
Upf, actually that's tricky. The ones in the DB are the ones **actually run**, but now we have the case of manual scripts which we never know when they were ran. Unless we actually put them in the DB after running, which I think would be the correct way to do it.
So then what do you think about manually registering a manually run script in the DB to keep consistency and clarity over what's in the DB?

Reply:
Oh well in that case I would stick to the DB collection as I find it sure to be correct. For example running the code locally would generate an incorrect hash because we don't run the migrations automatically locally.

Reply:
The manually run **are** put into the db when they have been run. It's tricky I agree, I just have the feeling that the importing systems are talking to each other via REST, so actually the schema you get is the schema of the running app (which in turn still has to be aligned to the db). I also thought it would be easier and would spare a db call as the migration names are already there in this file, you just would need to hash them. Anyway, I'm not insisting on this. I would propose though to remove the side effect of the field `schemaHash` just being set if one has not forgotten to call this function first. Maybe make the field "private" and always call `getSchemaHash` (the db call can still be executed only once)?

Reply:
We now run all migrations everytime (also locally) BTW

Reply:
Actually now that I think about it, yesterday when we ran the manual script we didn't restart the services. This means that the hash would be out of date for a long time possibly. 
I'd propose to either agree to restart the services each time we do a manual run or easier, to regenerate the hash each time we need it.

Reply:
True, in a normal scenario when scripts are run automatically, the service is started afterwards. So I think restart would be the consistent way for manual scripts.

I was running on old knowledge with how the scripts are run locally 😵‍💫 

So coming back to where to generate the hash from, having a manual script file does not indicate whether it was run or not, as it looks like the DB does indicate that. So let's stick to using the migrations collection then.

Reply:
OK, but can we please remove the side effect?

Reply:
Where would you think of putting this call 🤔 ? My thought was that it should always come after calling `runMigrationScripts`.

Reply:
Sounds good to me

Reply:
I wouldn't call it at all without any reason. We can create the hash the first time it is requested and then memorize it.

Reply:
Ok, I did not get that from the previous comment. Thanks, will do that.

Reply:
Done, please check if it's what you meant.

Reply:
It's what I meant. Thanks.